### PR TITLE
wr-287: Add ability to share articles with the registered users

### PR DESCRIPTION
**Pull request by drMangust**
I've done everything, but I don't like one thing (maybe I'm worried for nothing)

- I'm in the SharedArticleRoute wrapper, if there is a user, I do a token dispatch to the server and get the id
- then it doesn't matter what page we get to, whether on ArticlePage or SharedArticlePage, we again need to go to the server for page data (on ArticlePage, for comments)

### CODE REVIEW

File: frontend/src/libs/components/shared-article-route/shared-article-route.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  const dispatch = useAppDispatch();
+  const { user, articleId, articleIdByTokenDataStatus } = useAppSelector(
+    ({ auth, articles: { articleIdByToken, articleIdByTokenDataStatus } }) => ({
+      user: auth.user,
+      articleId: articleIdByToken,
+      articleIdByTokenDataStatus: articleIdByTokenDataStatus,
+    }),
+  );
+  const { token } = useParams();
+  const hasUser = Boolean(user);
```
Comment (IrynaK-a) [inline frontend/src/libs/components/shared-article-route/shared-article-route.tsx:31]: Could we just use the user in the next condition without creating a new variable?

Comment (drMangust) [reply frontend/src/libs/components/shared-article-route/shared-article-route.tsx:31]: I left it as it is done in ProtectedRoute and PublicRoute. If I had chosen, I wouldn't have created it

File: frontend/src/libs/components/shared-article-route/shared-article-route.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      articleIdByTokenDataStatus === DataStatus.REJECTED
+    );
+
+    if (!articleId && !isLoading) {
+      return <Navigate to={redirectPath} />;
+    }
+
+    return (
+      <Loader isLoading={isLoading} hasOverlay type="circular">
+        <Navigate to={`${AppRoute.ARTICLES}/${articleId}`} replace />
```
Comment (andanf-e) [inline frontend/src/libs/components/shared-article-route/shared-article-route.tsx]:
```suggestion
        <Navigate
          to={configureString(AppRoute.ARTICLES_$ID, { id: articleId }) as typeof AppRoute.ARTICLES_$ID;}
          replace
        />
```

Comment (drMangust) [reply frontend/src/libs/components/shared-article-route/shared-article-route.tsx]: [87f0669](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/303/commits/87f066940eac038f89dfc05478eebe3dfc78fe5d)

**Review by andanf-e — APPROVED** [review]

**Review by mhrechkin — APPROVED** [review]

**Review by NazarKovtun1 — APPROVED** [review]

**Pull request merged**